src/catalog/collection.rs

```diff
@@ -127,6 +138,45 @@ mod test {
     use rusqlite::params as sql_params;
     use serde_json::json;
 
+    #[test]
```

Comment:
A lot of repetitive fixture creation here. Really this is just about exercising these getters against _some_ fixture. How about folding the assertions themselves into `test_register` ?